packages/bitcore-node/src/models/transaction.ts

```diff
@@ -170,7 +171,9 @@ export class MongoWriteStream extends Transform {
   }
 
   async _transform(data: Array<any>, _, done) {
-    await this.collection.bulkWrite(data);
+    await Promise.all(
+      partition([...data], data.length / Config.get().maxPoolSize).map(batch => this.collection.bulkWrite(batch))
```

Comment:
i thought we were on slice, why does the data array need to be cloned?

Reply:
Oh nice, good catch